### PR TITLE
modules/ignition,platforms/*: Allow NFS configuration on CL nodes

### PR DESCRIPTION
**Pull request by lander2k2**
This will allow for the configuration of NFS to, for example, specify
versions of NFS protocol that may be used.  It is flexible enough to
allow any configuration desired by allowing the contents of the config
file to be passed in as a tfvar.

### CODE REVIEW

**Comment by cpanato** [comment]
retest this please

File: platforms/aws/main.tf [hunk 1]
```diff
@@ -156,6 +157,7 @@ module "masters" {
   ign_kubelet_service_id               = "${module.ignition_masters.kubelet_service_id}"
   ign_locksmithd_service_id            = "${module.ignition_masters.locksmithd_service_id}"
   ign_max_user_watches_id              = "${module.ignition_masters.max_user_watches_id}"
+  ign_nfs_config_id                    = "${length(var.tectonic_nfs_config) > 0 ? module.ignition_masters.nfs_config_id : ""}"
```
Comment (enxebre) [inline platforms/aws/main.tf]: Note that although this seems to work length is meant to me applied with list, not with strings https://www.terraform.io/docs/configuration/interpolation.html#length-list-

**Comment by enxebre** [comment]
lgtm

File: modules/ignition/assets.tf [hunk 2]
```diff
@@ -248,3 +248,21 @@ data "ignition_file" "systemd_default_env" {
     content = "${data.template_file.systemd_default_env.rendered}"
   }
 }
+
+data "template_file" "nfs_config" {
```
Comment (spangenberg) [inline modules/ignition/assets.tf]: unnecessary since the template file doesn't extend var.nfs_config

File: modules/ignition/assets.tf [hunk 3]
```diff
@@ -248,3 +248,21 @@ data "ignition_file" "systemd_default_env" {
...
+  template = "${file("${path.module}/resources/conf.d/nfs")}"
+}
+
+data "ignition_file" "nfs_config" {
+  path       = "/etc/conf.d/nfs"
+  mode       = 0644
+  filesystem = "root"
+
+  content {
+    content = "${data.template_file.nfs_config.rendered}"
```
Comment (spangenberg) [inline modules/ignition/assets.tf]: could use var.nfs_config directly

File: modules/ignition/outputs.tf [hunk 4]
```diff
@@ -163,3 +163,11 @@ output "systemd_default_env_id" {
 output "systemd_default_env_rendered" {
   value = "${data.template_file.systemd_default_env.rendered}"
 }
+
+output "nfs_config_id" {
+  value = "${data.ignition_file.nfs_config.id}"
+}
+
+output "nfs_config_rendered" {
+  value = "{data.template_file.nfs_config.rendered}"
```
Comment (spangenberg) [inline modules/ignition/outputs.tf]: could use var.nfs_config directly

**Comment by kalmog** [comment]
Also tested & passed: local vmware install

**Comment by cpanato** [comment]
retest this please. AWS/AZURE/GOVCLOUD

File: Documentation/variables/config.md [hunk 5]
```diff
@@ -41,6 +41,7 @@ This document gives an overview of variables used in all platforms of the Tecton
 | tectonic_license_path | The path to the tectonic licence file. You can download the Tectonic license file from your Account overview page at [1].<br><br>[1] https://account.coreos.com/overview<br><br>Note: This field MUST be set manually prior to creating the cluster unless `tectonic_vanilla_k8s` is set to `true`. | string | `` |
 | tectonic_master_count | The number of master nodes to be created. This applies only to cloud platforms. | string | `1` |
 | tectonic_networking | (optional) Configures the network to be used in Tectonic. One of the following values can be used:<br><br>- "flannel": enables overlay networking only. This is implemented by flannel using VXLAN.<br><br>- "canal": [ALPHA] enables overlay networking including network policy. Overlay is implemented by flannel using VXLAN. Network policy is implemented by Calico.<br><br>- "calico": [ALPHA] enables BGP based networking. Routing and network policy is implemented by Calico. Note this has been tested on baremetal installations only. | string | `flannel` |
+| tectonic_nfs_config | (optional) Content of NFS configuration file. Example: "OPTS_RPC_MOUNTD=\"8 -N 2 -N 3 -V 4 -V 4.1\"" | string | `` |
```
Comment (alexsomesan) [inline Documentation/variables/config.md]: We are usually trying really hard not to inline contents for files in input variables.
Doing so will make it complicated to process these files with tools, such as the cli tool. Is is also very confusing to look at when dealing with the varfile manually in a editor.

I suggest we require the desired NFS config in a file and make this variable a path to that file.
You can use the `"${file(...)}"` interpolation function in the templates to read the contents from the supplied file.

**Comment by cpanato** [comment]
all green, ready for review and merge

**Comment by sym3tri** [comment]
This was moved to https://github.com/coreos/tectonic-installer/pull/2815 and merged

**Pull request closed without merging**